packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.ts

```diff
@@ -0,0 +1,327 @@
+  logHoverCitation: (
+    citationId: string,
+    citationHoverTimeMs: number
+  ) => CustomAction;
+  logGeneratedAnswerShowAnswers: () => CustomAction;
+  logGeneratedAnswerHideAnswers: () => CustomAction;
+  logCopyGeneratedAnswer: () => CustomAction;
+  logRephraseGeneratedAnswer: (
+    responseFormat: GeneratedResponseFormat
+  ) => LegacySearchAction;
+  logRetryGeneratedAnswer: () => LegacySearchAction;
+}
+
+/**
+ * Creates a core `GeneratedAnswer` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `GeneratedAnswer` properties.
+ * @returns A `GeneratedAnswer` controller instance.
+ */
+export function buildCoreGeneratedAnswer(
+  engine: CoreEngine,
+  analyticsClient: GeneratedAnswerAnalyticsClient,
+  props: GeneratedAnswerProps = {}
+): GeneratedAnswer {
+  if (!loadGeneratedAnswerReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const {dispatch} = engine;
+  const controller = buildController(engine);
+  const getState = () => engine.state;
+
+  const isVisible = props.initialState?.isVisible;
+  if (isVisible !== undefined) {
+    dispatch(setIsVisible(isVisible));
+  }
+  const initialResponseFormat = props.initialState?.responseFormat;
+  if (initialResponseFormat) {
+    dispatch(updateResponseFormat(initialResponseFormat));
+  }
+
+  const fieldsToIncludeInCitations = props.fieldsToIncludeInCitations;
+  if (fieldsToIncludeInCitations) {
+    dispatch(registerFieldsToIncludeInCitations(fieldsToIncludeInCitations));
+  }
+
+  subscribeStateManager.subscribeToSearchRequests(engine);
+
+  return {
+    ...controller,
+
+    get state() {
+      return getState().generatedAnswer;
+    },
+
+    like() {
+      if (!this.state.liked) {
+        dispatch(likeGeneratedAnswer());
+        dispatch(analyticsClient.logLikeGeneratedAnswer());
+      }
+    },
+
+    dislike() {
+      if (!this.state.disliked) {
+        dispatch(dislikeGeneratedAnswer());
+        dispatch(analyticsClient.logDislikeGeneratedAnswer());
+      }
+    },
+
+    openFeedbackModal() {
+      dispatch(openGeneratedAnswerFeedbackModal());
+    },
+
+    closeFeedbackModal() {
+      dispatch(closeGeneratedAnswerFeedbackModal());
+    },
+
+    sendFeedback(feedback) {
+      dispatch(analyticsClient.logGeneratedAnswerFeedback(feedback));
+      dispatch(sendGeneratedAnswerFeedback());
+    },
+
+    sendDetailedFeedback(details) {
+      dispatch(analyticsClient.logGeneratedAnswerDetailedFeedback(details));
+      dispatch(sendGeneratedAnswerFeedback());
+    },
+
+    logCitationClick(citationId: string) {
+      dispatch(analyticsClient.logOpenGeneratedAnswerSource(citationId));
+    },
+
+    logCitationHover(citationId: string, citationHoverTimeMs: number) {
+      dispatch(
+        analyticsClient.logHoverCitation(citationId, citationHoverTimeMs)
+      );
+    },
+
+    rephrase(responseFormat: GeneratedResponseFormat) {
+      dispatch(updateResponseFormat(responseFormat));
```

Comment:
Why aren't we dispatching an executeSearch here? 
Because the search is different in insight than with search?

Should we have like a "SearchClient" just like the analytics client that would provide the methods to call to execute search etc.?

Reply:
Exactly we are not dispatching an executeSearch here because the search is different in insight than with search.

I'm not having a SearchClient  because currently for example for the rephrase method we do:
#### For search: 
```
        executeSearch({
          legacy:
            generatedAnswerAnalyticsClient.logRephraseGeneratedAnswer(
              responseFormat
            ),
        })
```
#### For insight:
````
        executeSearch(
          generatedAnswerInsightAnalyticsClient.logRetryGeneratedAnswer()
        )
````
So even the analytics action passed to the executeSearch is different, one is using the key `legacy` and one is not.
Same thing with the `Retry` method in this controlle.

That's why I thought that it would be better and more simple to avoid using a search client and define the methods separately in the two controllers.

Probably when we standardize how the analytics should be logged with the new protocol, it would be much easier to use a SearchClient exactly like you mentioned.